lib/src/network/protocol/pipe.rs

```diff
@@ -249,11 +253,14 @@ impl<Front:SocketHandler> Pipe<Front> {
         self.readiness.back_interest.remove(Ready::writable());
         return ClientResult::Continue;
       }
-      let (current_sz, current_res) = self.backend.socket_write(self.front_buf.next_output_data());
-      socket_res = current_res;
-      self.front_buf.consume_output_data(current_sz);
-      self.front_buf_position += current_sz;
-      sz += current_sz;
+
+      if let Some(ref mut backend) = self.backend {
```

Comment:
this should be moved outside of the loop. If we reach this part and for some reason, backend is `None`, this will make an infinite loop

Reply:
Fixed